driver/test/column_bindings_it.cpp

```diff
@@ -0,0 +1,487 @@
+    std::vector<SQLDOUBLE> col3(array_size);
+    std::vector<SQLLEN> col3_ind(array_size);
+
+    ODBC_CALL_ON_STMT_THROW(hstmt,
+        SQLBindCol(
+            hstmt,
+            3,
+            getCTypeFor<std::decay_t<decltype(col3[0])>>(),
+            adjsut_ptr(&col3[0]),
+            sizeof(col3[0]),
+            adjsut_ind_ptr(&col3_ind[0])
+        )
+    );
+
+    std::vector<FixedStringBuffer<SQLCHAR, 41>> col4(array_size);
+    std::vector<SQLLEN> col4_ind(array_size);
+
+    ODBC_CALL_ON_STMT_THROW(hstmt,
+        SQLBindCol(
+            hstmt,
+            4,
+            getCTypeFor<SQLCHAR *>(),
+            adjsut_ptr(&col4[0]),
+            sizeof(col4[0].data) * sizeof(col4[0].data[0]),
+            adjsut_ind_ptr(&col4_ind[0])
+        )
+    );
+
+    std::vector<SQLUBIGINT> col5(array_size);
+    std::vector<SQLLEN> col5_ind(array_size);
+
+    ODBC_CALL_ON_STMT_THROW(hstmt,
+        SQLBindCol(
+            hstmt,
+            5,
+            getCTypeFor<std::decay_t<decltype(col5[0])>>(),
+            adjsut_ptr(&col5[0]),
+            sizeof(col5[0]),
+            adjsut_ind_ptr(&col5_ind[0])
+        )
+    );
+
+    std::vector<SQLREAL> col6(array_size);
+    std::vector<SQLLEN> col6_ind(array_size);
+
+    ODBC_CALL_ON_STMT_THROW(hstmt,
+        SQLBindCol(
+            hstmt,
+            6,
+            getCTypeFor<std::decay_t<decltype(col6[0])>>(),
+            adjsut_ptr(&col6[0]),
+            sizeof(col6[0]),
+            adjsut_ind_ptr(&col6_ind[0])
+        )
+    );
+
+    std::size_t total_rows = 0;
+
+    while (true) {
+        // Must not reallocate the underlying buffers.
+
+        std::fill(col1.begin(), col1.end(), -123);
+        std::fill(col1_ind.begin(), col1_ind.end(), -123);
+
+        std::fill(col2.begin(), col2.end(), FixedStringBuffer<SQLCHAR, 31>{});
+        std::fill(col2_ind.begin(), col2_ind.end(), -123);
+
+        std::fill(col3.begin(), col3.end(), -123);
+        std::fill(col3_ind.begin(), col3_ind.end(), -123);
+
+        std::fill(col4.begin(), col4.end(), FixedStringBuffer<SQLCHAR, 41>{});
+        std::fill(col4_ind.begin(), col4_ind.end(), -123);
+
+        std::fill(col5.begin(), col5.end(), 123456789);
+        std::fill(col5_ind.begin(), col5_ind.end(), -123);
+
+        std::fill(col6.begin(), col6.end(), -123);
+        std::fill(col6_ind.begin(), col6_ind.end(), -123);
+
+        rows_processed = 0;
+
+        const SQLRETURN rc = SQLFetch(hstmt);
+
+        if (rc == SQL_NO_DATA)
+            break;
+
+        if (rc == SQL_ERROR)
+            throw std::runtime_error(extract_diagnostics(hstmt, SQL_HANDLE_STMT));
+
+        if (rc == SQL_SUCCESS_WITH_INFO)
+            std::cout << extract_diagnostics(hstmt, SQL_HANDLE_STMT) << std::endl;
+
+        if (!SQL_SUCCEEDED(rc))
+            throw std::runtime_error("SQLFetch return code: " + std::to_string(rc));
+
+        ASSERT_LE(rows_processed, array_size);
+
+        for (std::size_t i = 0; i < array_size; ++i) {
```

Comment:
Same thing here: there are some not so easily guessable assumptions about results.

Reply:
Added comments about what is happening in general.